### DISCUSSION TITLE
No ESP Box Question

### DISCUSSION THREAD

**Discussion by asktimfisher** [toverainc/willow#53] [Q&A]
Can Willow be setup and used without an ESP Box? (I'll buy an ESP Box to support!). I'd like to play with this with different hardware.

My setup includes HA on a NUC. I'd love to install Willow as another docker container on that NUC and hook Willow up to a wall-mount tablet I have in my living room and see if I can get audio streaming from there to the Willow server and voice responses back to it.

Lots of stuff to figure out myself in that explanation, but really curious about the first one.

**Comment by kristiankielhofner** [comment]
An interesting thing about Willow is we don't have any direct involvement in the ESP BOX. Espressif makes them and sells them through distributors. Until you buy one (wherever) and flash with Willow we don't have anything to do with it!

Think of Willow like OpenWRT. You buy a TP-Link router from wherever and flash OpenWRT on it because it's better than factory firmware and often even more expensive proprietary commercial solutions. That's what Willow is doing for voice interfaces.

Willow has a very specific mission and goal: no one should ever have to compromise and put big tech microphones in their environment to interact with technology via voice. Willow is laser-focused on matching or beating Echo, Google Home, etc by every possible metric and then some - quality, reliability, accuracy, flexibility, privacy, user control, integrations, price, trust, and more.

Willow is able to achieve these goals by using hardware and software purposefully and carefully engineered from the plastic on the enclosure to the text that comes out of speech recognition. Echo was purposefully engineered the same way. Same with Google Home.

There are a lot of fundamentals about acoustically tuned enclosures, microphone cavities, signal processing, etc but I won't bore you with all of that.

From what you're describing it really sounds like you're looking for something else - software to run on an existing tablet and a NUC.

**Comment by jonsmirl** [comment]
The ESP32 S3 Box is a dev kit from Espressif. There is nothing special about it and it is completely open source.
https://github.com/espressif/esp-box/blob/master/hardware/esp32_s3_box_v2.5/schematic/SCH_ESP32-S3-BOX_V2.5_20211011.pdf
The advantage to using it is that it already has a working wakeword engine. There is nothing stopping anyone from taking the open source schematics and layout and using them to make different versions. The ESP32 S3 Box is a little expensive because it is hand assembled and the case is 3D printed.  Making it in Q1000 volume would cost about $25 plus $10K in NRE.

You can probably talk these guys into making a custom model for you.
https://www.makerfabs.com/matouch.html
